java/src/main/native/ai_onnxruntime_OrtSession.c

```diff
@@ -13,227 +13,286 @@
+      break;
+    }
+    jstring name = (*jniEnv)->NewStringUTF(jniEnv, outputName);
+    (*jniEnv)->SetObjectArrayElement(jniEnv, array, i, name);
+    code = checkOrtStatus(jniEnv, api, api->AllocatorFree(allocator, outputName));
+    if (code != ORT_OK) {
+      // break out on error, return array and let Java throw the exception.
+      break;
     }
+  }
 
-    return array;
+  return array;
 }
 
 /*
  * Class:     ai_onnxruntime_OrtSession
  * Method:    getInputInfo
  * Signature: (JJJ)[Lai/onnxruntime/NodeInfo;
  */
-JNIEXPORT jobjectArray JNICALL Java_ai_onnxruntime_OrtSession_getInputInfo
-  (JNIEnv * jniEnv, jobject jobj, jlong apiHandle, jlong sessionHandle, jlong allocatorHandle) {
-    (void) jobj; // Required JNI parameter not needed by functions which don't need to access their host object.
-    const OrtApi* api = (const OrtApi*) apiHandle;
-    OrtAllocator* allocator = (OrtAllocator*) allocatorHandle;
-
-    // Setup
-    char *nodeInfoClassName = "ai/onnxruntime/NodeInfo";
-    jclass nodeInfoClazz = (*jniEnv)->FindClass(jniEnv, nodeInfoClassName);
-    jmethodID nodeInfoConstructor = (*jniEnv)->GetMethodID(jniEnv,nodeInfoClazz, "<init>", "(Ljava/lang/String;Lai/onnxruntime/ValueInfo;)V");
-
-    // Get the number of inputs
-    size_t numInputs = Java_ai_onnxruntime_OrtSession_getNumInputs(jniEnv, jobj, apiHandle, sessionHandle);
-
-    // Allocate the return array
-    jobjectArray array = (*jniEnv)->NewObjectArray(jniEnv,safecast_size_t_to_jsize(numInputs),nodeInfoClazz, NULL);
-    for (size_t i = 0; i < numInputs; i++) {
-        // Read out the input name and convert it to a java.lang.String
-        char* inputName;
-        checkOrtStatus(jniEnv,api,api->SessionGetInputName((OrtSession*)sessionHandle, i, allocator, &inputName));
-        jstring name = (*jniEnv)->NewStringUTF(jniEnv,inputName);
-        checkOrtStatus(jniEnv,api,api->AllocatorFree(allocator,inputName));
-
-        // Create a ValueInfo from the OrtTypeInfo
-        OrtTypeInfo* typeInfo;
-        checkOrtStatus(jniEnv,api,api->SessionGetInputTypeInfo((OrtSession*)sessionHandle, i, &typeInfo));
-        jobject valueInfoJava = convertToValueInfo(jniEnv,api,typeInfo);
-        api->ReleaseTypeInfo(typeInfo);
-
-        // Create a NodeInfo and assign into the array
-        jobject nodeInfo = (*jniEnv)->NewObject(jniEnv, nodeInfoClazz, nodeInfoConstructor, name, valueInfoJava);
-        (*jniEnv)->SetObjectArrayElement(jniEnv, array,safecast_size_t_to_jsize(i),nodeInfo);
+JNIEXPORT jobjectArray JNICALL Java_ai_onnxruntime_OrtSession_getInputInfo(JNIEnv* jniEnv, jobject jobj, jlong apiHandle, jlong sessionHandle, jlong allocatorHandle) {
+  (void)jobj;  // Required JNI parameter not needed by functions which don't need to access their host object.
+  const OrtApi* api = (const OrtApi*)apiHandle;
+  OrtSession* session = (OrtSession*)sessionHandle;
+  OrtAllocator* allocator = (OrtAllocator*)allocatorHandle;
+
+  // Setup
```

Comment:
>// Setu [](http://example.com/codeflow?start=2&length=7)

The information here is not necessary until the very end of the function. It would be more comprehensible to move it there./